### DISCUSSION TITLE
Mutation is not understood by Pluto.jl

### DISCUSSION THREAD

**Discussion by samuela** [fonsp/Pluto.jl#1329] [Pluto development]
Pluto.jl does not seem to always show the correct output when mutation is involved between the cells. For example, https://user-images.githubusercontent.com/226872/126718376-37fe08fc-ddc8-4e5e-a985-6f4502a66d30.mov

AFAICT there are two bugs captured in the screencast:
1. Around the 12 second mark, the third cell (`x`) is evaluated, but the correct value is not returned. It still shows as if `x == [3]`.
2. Starting around the 20 second mark, I evaluate the `push!(x, 3)` cell multiple times adding a bunch of 3s to `x`. But the third cell is not re-evaluated. It does not display the correct value until being manually re-evaluated.

Personally, I find the greatest thing about Pluto over Jupyter/etc to be that Pluto "solves" the is-this-cell-stale question. Unfortunately however this example shows me that I cannot always trust Pluto on that promise.

Is addressing mutation something that Pluto.jl plans to cover in the future? I imagine that mutation makes the cell ordering problem more difficult. But even a warning as simple as 

> "Hey, it looks like you mutated some value outside of what Pluto understands. Try keeping all mutation in a single cell with `begin ... end` blocks instead!" 

would be super helpful.

**Comment by fonsp** [comment]
More discussion in #564
